odl/solvers/functional/default_functionals.py

```diff
@@ -324,8 +332,8 @@ def convex_conj(self):
 
     def __repr__(self):
         """Return ``repr(self)``."""
-        return '{}({!r}, exponent={})'.format(self.__class__.__name__,
-                                              self.domain,
+        return '{}({!r} -> {!r}, exponent={})'.format(self.__class__.__name__,
+                                              self.domain, self.range,
```

Comment:
We try, whenever possible, to make `repr` return a string that is a valid Python expression. So please add `range` as a second argument.
Note that there's a quite huge overhaul of all this repr stuff going on, see #1276 for reference. You don't need to spend too much effort on this part.

Reply:
Actually, after looking at the failing doctest (output `L2Norm(rn(3) -> RealNumbers())` doesn't match expected output `L2Norm(rn(3))`, something like this), I'd suggest not changing anything in `__repr__`. That's suboptimal for the non-standard case but at least doesn't break the printout for the default case.